src/components/Personalization/createOnClickHandler.js

```diff
@@ -30,7 +31,18 @@ export default ({
       );
 
       if (isNonEmptyArray(decisionsMeta)) {
-        event.mergeXdm({ eventType: INTERACT });
+        const xdm = { eventType: INTERACT };
+        const scope = decisionsMeta[0].scope;
+
+        if (scope !== PAGE_WIDE_SCOPE) {
+          xdm.web = {
```

Comment:
Wouldn't this replace fully replace `xdm.web`? Should we be merging in `viewName: scope` instead?

Let's add this test case maybe, where we add `viewName` to an existing `web.webPageDetails`.

Reply:
`event.mergeXDM` is using `deepAssign`, so it is safe to do so.